project_sequence/models/project_project.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+
+from odoo import api, fields, models
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    code = fields.Char(
+        string="Number sequence",
+        readonly=True,
+        copy=False,
+        default=lambda self: self.env["ir.sequence"].next_by_code("project.sequence"),
+    )
+    name = fields.Char(required=False)
+    display_name = fields.Char(compute="_compute_display_name_project")
```

Comment:
You can make it stored and indexed, so users can search on the display name directly:

```suggestion
    display_name = fields.Char(compute="_compute_display_name_project", store=True, index=True)
```